http/src/lib.rs

```diff
@@ -553,11 +593,12 @@ fn serve<M: jsonrpc::Metadata, S: jsonrpc::Middleware<M>>(
 							keep_alive,
 						);
 
-						http.serve_connection(socket, service)
-							.map_err(|e| error!("Error serving connection: {:?}", e))
-							.then(|_| Ok(()))
+						tokio::spawn(
+							http.serve_connection(socket, service)
+								.map_err(|e| error!("Error serving connection: {:?}", e))
```

Comment:
rustfmt is angry with you about this line. Maybe put the `map_err(|e| {` on the preceding line?